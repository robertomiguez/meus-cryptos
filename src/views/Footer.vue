<template>
    <div class="footer">
        <div class="follow">
            Powered by CoinCap API | Follow me:
            <font-awesome-icon
              :icon="{ prefix: 'fab', iconName: 'linkedin' }"
              @click="openUrl('https://linkedin.com/in/miguezr')"
              class='icon'/>
        </div>
    </div>
</template>

<script>
export default {
  name: 'Footer',
  methods: {
    openUrl (url) {
      window.open(url, '_blank')
    }
  }
}
</script>

<style scoped>
.footer {
   position: fixed;
   bottom: 0;
   width: 100%;
   text-align: center;
   height: 30px;

}
.follow {
    padding: 5px 15px 0 0;
}
.icon {
    cursor: pointer;
    padding-left: 7px;
}
</style>
