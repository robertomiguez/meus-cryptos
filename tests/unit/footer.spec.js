import { shallowMount } from '@vue/test-utils'
import Footer from '@/views/Footer'

describe('Footer', () => {
  const build = () => {
    const wrapper = shallowMount(Footer, {
      stubs: ['font-awesome-icon']
    })

    return {
      wrapper
    }
  }

  it('renders the component', () => {
    // arrange
    const { wrapper } = build()
    // assert
    expect(wrapper.html()).toMatchSnapshot()
  })
})
