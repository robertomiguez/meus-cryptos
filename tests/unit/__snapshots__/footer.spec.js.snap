// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders the component 1`] = `
<div class="footer">
  <div class="follow">
    Powered by CoinCap API | Follow me:
    <font-awesome-icon-stub icon="[object Object]" class="icon"></font-awesome-icon-stub>
  </div>
</div>
`;
